test/infrastructure/docker/exp/internal/controllers/dockermachinepool_controller_phases.go

```diff
@@ -189,36 +180,34 @@ func (r *DockerMachinePoolReconciler) reconcileDockerMachines(ctx context.Contex
 
 	// For each DockerMachine, fetch the owner Machine and copy the clusterv1.DeleteMachineAnnotation to the DockerMachine if it exists before sorting the DockerMachines.
 	// This is done just before sorting to guarantee we have the latest copy of the Machine annotations.
-	readyMachines, err := r.getReadyMachines(ctx, cluster, machinePool, dockerMachinePool)
-	if err != nil {
-		return err
-	}
-	outdatedMachines, err := r.getMachinesWithoutLatestModel(ctx, cluster, machinePool, dockerMachinePool, labelFilters)
+	dockerMachinesWithAnnotation, err := r.propagateMachineDeleteAnnotation(ctx, dockerMachineSet)
 	if err != nil {
 		return err
 	}
 
-	// For each DockerMachine, fetch the owner Machine and copy the clusterv1.DeleteMachineAnnotation to the DockerMachine if it exists before sorting the DockerMachines.
-	readyMachinesWithAnnotation, err := r.propagateMachineDeleteAnnotation(ctx, readyMachines)
-	if err != nil {
-		return err
+	// Sort DockerMachines with the clusterv1.DeleteMachineAnnotation to the front of each list.
+	// If providers already have a sorting order for instance deletion, i.e. oldest first or newest first, the clusterv1.DeleteMachineAnnotation must take priority.
+	// For example, if deleting by oldest, we expect the InfraMachines with clusterv1.DeleteMachineAnnotation to be deleted first followed by the oldest, and the second oldest, etc.
+	orderedDockerMachines := orderByDeleteMachineAnnotation(dockerMachinesWithAnnotation)
+
+	numReadyMachines := 0
+	for i := range orderedDockerMachines {
+		dockerMachine := orderedDockerMachines[i]
+		if dockerMachine.Status.Ready || conditions.IsTrue(&dockerMachine, clusterv1.ReadyCondition) {
+			numReadyMachines++
```

Comment:
this is confusing because `numReadyMachines` != `len(readyMachines)`

As far as I understand, ready machines is a subset of all "ready" machines (only the ones that are not outdated). Consider renaming the variables to reflect that.